include/zephyr/drivers/clock_control/clock_control_mcux_ccm_rev3.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_MCUX_CCM_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_MCUX_CCM_H_
+
+#include <stdint.h>
+#include <zephyr/device.h>
+
+#define IMX_CCM_MAX_SOURCES 4
+
+#define IMX_CCM_FIXED_FREQ(src) ((src).source.fixed.freq)
+#define IMX_CCM_PLL_MAX_FREQ(src) ((src).source.pll.max_freq)
+#define IMX_CCM_RATE_LIMIT(src) \
+	((src).type == IMX_CCM_TYPE_FIXED ? \
+	 IMX_CCM_FIXED_FREQ(src) : IMX_CCM_PLL_MAX_FREQ(src))
+
+enum imx_ccm_clock_state {
+	IMX_CCM_CLOCK_STATE_INIT = 0, /* initial state - clock may be gated or not */
+	IMX_CCM_CLOCK_STATE_GATED, /* clock is gated */
+	IMX_CCM_CLOCK_STATE_UNGATED, /* clock is not gated */
+};
+
+enum imx_ccm_type {
+	IMX_CCM_TYPE_FIXED = 1,
+	IMX_CCM_TYPE_PLL,
+	IMX_CCM_TYPE_MAX,
+};
+
+struct imx_ccm_fixed {
+	char *name;
```

Comment:
This is primarily used in the clock tree description for readability. It might be easier to understand it if the clock is also assigned a string name and not just an ID from the NXP HAL.

Reply:
I'm wondering if we can use macros defined in a header file for the clock ID values. This way we avoid the overhead of the name constants being stored, but the clock tree defined in the dts file is still using more friendly clock names